psi4/src/psi4/lib3index/fittingmetric.cc

```diff
@@ -387,13 +380,12 @@ void FittingMetric::form_QR_inverse(double tol) {
             }
 
         // First, find out how much workspace to provide
-        C_DORGQR(n, n, n, Qp[0], n, tau, &work_size, -1);
+        C_DORGQR(n, n, n, Qp[0], n, tau.data(), &work_size, -1);
 
         // Now, form Q
         lwork = (int)work_size;
-        work = new double[lwork];
-        C_DORGQR(n, n, n, Qp[0], n, tau, work, lwork);
-        delete[] work;
+        work.resize(lwork);
```

Comment:
`work` is already of size `lwork`, so this line isn't necessary.

Reply:
`lwork` gets resized from `C_DORGQR(n, n, n, Qp[0], n, tau.data(), &work_size, -1);` if I'm not mistaken.

Reply:
That's right. In that case, please update the comment above the line you quoted to explicitly say it writes the optimal length into `work_size`. I would not have guessed that without checking to see what exactly that function does.